### PR TITLE
Fix 5 broken links

### PR DESCRIPTION
**Pull request by CaitinChen**
@YiniXu9506 PTAL

### CODE REVIEW

**Comment by husiyu** [comment]
@shenli PTAL

File: tidb-source-code-reading-3.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ SQL 有很多种，比如读、写、修改、删除以及管理类的 SQL，每
 
 + [Session](https://github.com/pingcap/tidb/blob/source-code/session.go#L62)
 
-+ [RecordSet](https://github.com/pingcap/tidb/blob/master/ast/ast.go#L136)
++ [RecordSet](https://github.com/pingcap/tidb/blob/rc2.3/ast/ast.go#L136)
```
Comment (shenli) [inline tidb-source-code-reading-3.md]: https://github.com/pingcap/tidb/blob/source-code/ast/ast.go#L136

File: tidb-source-code-reading-3.md [hunk 2]
```diff
@@ -184,7 +184,7 @@ Session 中最重要的函数是 [Execute](https://github.com/pingcap/tidb/blob/
 ```
 
 
-这个结构实现了 [`ast.RecordSet`](https://github.com/pingcap/tidb/blob/master/ast/ast.go#L142) 接口，从字面上大家可以看出，这个接口代表了查询结果集的抽象，我们看一下它的几个方法：
+这个结构实现了 [`ast.RecordSet`](https://github.com/pingcap/tidb/blob/rc2.3/ast/ast.go#L142) 接口，从字面上大家可以看出，这个接口代表了查询结果集的抽象，我们看一下它的几个方法：
```
Comment (shenli) [inline tidb-source-code-reading-3.md]: https://github.com/pingcap/tidb/blob/source-code/ast/ast.go#L136

File: tidb-source-code-reading-5.md [hunk 3]
```diff
@@ -244,7 +244,7 @@ func (s *Scanner) Errors() []error {
 ```
 
 
-另外 `lexer` 使用了 `字典树` 技术进行 `token` 识别，具体的实现代码在 [parser/misc.go](https://github.com/pingcap/tidb/blob/master/parser/misc.go)
+另外 `lexer` 使用了 `字典树` 技术进行 `token` 识别，具体的实现代码在 [parser/misc.go](https://github.com/pingcap/tidb/blob/rc2.3/parser/misc.go)
```
Comment (shenli) [inline tidb-source-code-reading-5.md]: https://github.com/pingcap/tidb/blob/source-code/parser/misc.go

File: tidb-source-code-reading-5.md [hunk 4]
```diff
@@ -315,7 +315,7 @@ type Visitor interface {
 }
 ```
 
-例如 [plan.preprocess](https://github.com/pingcap/tidb/blob/master/plan/preprocess.go) 是对 `AST`  做预处理，包括合法性检查以及名字绑定。
+例如 [plan.preprocess](https://github.com/pingcap/tidb/blob/rc2.3/plan/preprocess.go) 是对 `AST` 做预处理，包括合法性检查以及名字绑定。
```
Comment (shenli) [inline tidb-source-code-reading-5.md]: https://github.com/pingcap/tidb/blob/source-code/plan/preprocess.go

File: tidb-source-code-reading-5.md [hunk 5]
```diff
@@ -489,7 +489,7 @@ type SelectStmt struct {
 bin/goyacc -o parser/parser.go parser/parser.y 2>&1
 ```
 
-TiDB 对 `lexer` 和 `parser.go` 进行了封装，对外提供 [parser.yy_parser](https://github.com/pingcap/tidb/blob/master/parser/yy_parser.go)  进行 SQL 语句的解析：
+TiDB 对 `lexer` 和 `parser.go` 进行了封装，对外提供 [parser.yy_parser](https://github.com/pingcap/tidb/blob/rc2.3/parser/yy_parser.go)  进行 SQL 语句的解析：
```
Comment (shenli) [inline tidb-source-code-reading-5.md]: https://github.com/pingcap/tidb/blob/source-code/plan/preprocess.go

**Comment by CaitinChen** [comment]
@husiyu Please help merge. Thanks~

**Comment by husiyu** [comment]
LGTM

**Comment by queenypingcap** [comment]
@shenli  Please grant @CaitinChen the privilege to merge PR.

**Comment by YiniXu9506** [comment]
@queenypingcap This PR LGTM. I can merge it.

**Pull request merged**